### PR TITLE
Identity README

### PR DESCRIPTION
**Pull request by g2vinay**

### CODE REVIEW

File: sdk/identity/azure-identity/README.md [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+import com.azure.security.keyvault.secrets.SecretClient;
+
+DefaultAzureCredential defaultCredential = new DefaultAzureCredential();
+
+// Azure SDK client builders accept the credential as a parameter
+
+SecretClient client = SecretClient.builder()
+    .endpoint("https://{YOUR_VAULT_NAME}.vault.azure.net")
+    .credential(credential)
+    .build();
```
Comment (JonathanGiles) [inline sdk/identity/azure-identity/README.md]: Use the 'new' API (that hasn't yet been merged).

File: sdk/identity/azure-identity/README.md [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+// authenticate with client secret,
+ClientSecretCredential clientSecretCredential = new ClientSecretCredential()
+	    .clientId("<YOUR_CLIENT_ID>")
+	    .clientSecret("<YOUR_CLIENT_SECRET>")
+	    .tenantId("<YOUR_TENANT_ID>");
+
+KeyClient client = KeyClient.builder()
+    .endpoint("https://{YOUR_VAULT_NAME}.vault.azure.net")
+    .credential(clientSecretCredential)
+    .build();
```
Comment (JonathanGiles) [inline sdk/identity/azure-identity/README.md]: Same here

File: sdk/identity/azure-identity/README.md [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+
+// the chain can be used anywhere a credential is required
+
+// The fully qualified host name for the Event Hubs namespace. This is likely to be similar to:
+// {your-namespace}.servicebus.windows.net
+String host = "<< EVENT HUBS HOST >>"
+String eventHubPath = "<< NAME OF THE EVENT HUB >>";
+EventHubClient client = new EventHubClientBuilder()
+    .credential(host, eventHubPath, credentialChain)
+    .build();
```
Comment (JonathanGiles) [inline sdk/identity/azure-identity/README.md]: This is buildAsyncClient() now I believe.

File: sdk/identity/azure-identity/README.md [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+
+ClientSecretcredential firstServicePrincipal = new ClientSecretCredential()
+	    .clientId("<YOUR_CLIENT_ID>")
+	    .clientSecret("<YOUR_CLIENT_SECRET>")
+	    .tenantId("<YOUR_TENANT_ID>");
+
+ClientSecretcredential secondServicePrincipal = new ClientSecretCredential()
+	    .clientId("<YOUR_CLIENT_ID>")
+	    .clientSecret("<YOUR_CLIENT_SECRET>")
+	    .tenantId("<YOUR_TENANT_ID>");
```
Comment (schaabs) [inline sdk/identity/azure-identity/README.md]: I would use two different authentication types here, such as ManagedIdentityCredential followed by ClientSecretCredential, I think it demonstrates a more common use case than two different ClientSecretCredential instances.

File: sdk/identity/azure-identity/README.md [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+## Troubleshooting
+Credentials raise exceptions when they fail to authenticate. `ClientAuthenticationException` has a `message` attribute which
+describes why authentication failed. When raised by `ChainedTokenCredential`, the message collects error messages from each credential in the chain.
+
+## Next steps
+Currently the following client libraries support authenticating with `TokenCredential` and the Azure Identity library.  You can learn more about their use, and find additional documentation on use of these client libraries along samples with can be found in the links below.
+
+- [azure-eventhubs][eventhubs_client_library]
+- [azure-keyvault-keys][keys_client_library]
+- [azure-keyvault-secrets][secrets_client_library]
```
Comment (schaabs) [inline sdk/identity/azure-identity/README.md]: shouldn't this list contains azure-storage-blobs and azure-storage-queues as well?

File: sdk/identity/azure-identity/README.md (hunk 3 above)
Comment (g2vinay) [reply sdk/identity/azure-identity/README.md]: updated

File: sdk/identity/azure-identity/README.md (hunk 1 above)
Comment (g2vinay) [reply sdk/identity/azure-identity/README.md]: not part of this release.

File: sdk/identity/azure-identity/README.md (hunk 2 above)
Comment (g2vinay) [reply sdk/identity/azure-identity/README.md]: not part of this release.

File: sdk/identity/azure-identity/README.md (hunk 4 above)
Comment (g2vinay) [reply sdk/identity/azure-identity/README.md]: Updated.

File: sdk/identity/azure-identity/README.md (hunk 5 above)
Comment (g2vinay) [reply sdk/identity/azure-identity/README.md]: For Java, only storage-blob was released.
azure-storage-blob does support Azure Identity, but they currently lack samples demonstrating Azure Identity use case for auth. Once they are added, I will update the README here.

**Pull request merged**